src/substantial/src/backends/fs.rs

```diff
@@ -0,0 +1,177 @@
+use super::Backend;
+use crate::protocol::{
+    events::{Event, Records},
+    metadata::Metadata,
+};
+use anyhow::{Context, Ok, Result};
+use chrono::{DateTime, Utc};
+use protobuf::Message;
+use std::io::{Read, Write};
+use std::path::PathBuf;
+use std::{fs, path::Path};
+
+pub struct FsBackend {
+    root: PathBuf,
+}
+
+impl FsBackend {
+    pub fn new(root: &Path) -> Self {
+        for d in &["runs", "schedules", "leases"] {
+            fs::create_dir_all(root.join(d))
+                .with_context(|| "Failed to create directory")
+                .unwrap();
+        }
+        Self {
+            root: root.to_owned(),
```

Comment:
You might notice the missing `leases` on the backends, ressource management is done on the typegate and deno workers does most of the heavy lifting for channeling the events in a native non-blocking way (scheduling is done out of the box). If there are some use cases that I am missing please do tell me.

Reply:
Reintroducing it wouldn't be that difficult I imagine but it would imply spawning our own task queue, and run an infinite async loop that triggers `Start` for leased workflows but I am not really sure were such loop would be placed at or when such loop would be initialized (in a custom worker maybe?).

Reply:
Marking this as resolved, will extend the `WorkerManager` to handle leases.